### PR TITLE
Add -v flag to toggle compiler stdout, update examples

### PR DESCRIPTION
**Pull request by bobheadxi**
addon for branch `compiler`

```bash
bobbook:durian robertlin$ ./durian -c `pwd`/example.v0.dur
Compiling only...
Found filepath /Users/robertlin/Projects/durian/example.v0.dur
bobbook:durian robertlin$ ./durian -c -v `pwd`/example.v0.dur
Compiling only...
Found filepath /Users/robertlin/Projects/durian/example.v0.dur
02 01 00 00 00 00 00 00 00 F0
```

### CODE REVIEW

**Comment by rwblickhan** [comment]
LGTM. I believe Travis is failing because this branch calls memcpy but doesn't include cstring; this should be present in the other pull request, so we should probably merge that one first and then rebase this on top, and everything should pass.

File: src/main.cpp [hunk 1]
```diff
@@ -21,19 +22,23 @@ class ArgParser {
...
     ArgParser(int argc, char **argv)
-            : mode(Mode::None), optionalFilepath() {
+            : verbose(false), mode(Mode::None), optionalFilepath() {
         for (int i = 0; i < argc; i++) {
             args.emplace_back(argv[i]);
         }
         if (optionExists("-h") || optionExists("--help")) {
             printHelpAndExit(0);
         }
+        if (optionExists("-v")) {
```
Comment (rwblickhan) [inline src/main.cpp]: This could be simplified to `this->verbose = optionExists("-v")`.

File: src/main.cpp [hunk 2]
```diff
@@ -21,19 +22,23 @@ class ArgParser {
...
+            this->verbose = true;
+        }
         setModeIfExistsFlag("-p", Mode::PrettyPrint);
         setModeIfExistsFlag("-c", Mode::CompileOnly);
         setModeIfExistsFlag("-b", Mode::InterpretOnly);
         if (args.back() != "-p" &&
             args.back() != "-c" &&
             args.back() != "-b" &&
+            args.back() != "-v" &&
```
Comment (rwblickhan) [inline src/main.cpp:41]: We should probably think up a more principled way to do this in the future.

File: src/main.cpp [hunk 3]
```diff
@@ -123,7 +128,7 @@ int prettyPrint(std::pair<bool, std::string> optionalFilepath) {
     }
 }
 
-int compile(std::pair<bool, std::string> optionalFilepath) {
+int compile(std::pair<bool, std::string> optionalFilepath, bool verbose) {
```
Comment (rwblickhan) [inline src/main.cpp:131]: Do you mind adding the `verbose` option to the other functions too? I suspect we'll want it there too.

File: src/main.cpp [hunk 4]
```diff
@@ -148,9 +153,13 @@ int compile(std::pair<bool, std::string> optionalFilepath) {
         std::ofstream outFile("tmp.durb", std::ios::binary);
         std::copy(outBytecode.cbegin(),
                   outBytecode.cend(),
-                  std::ostreambuf_iterator(outFile));
-        for (unsigned char bytecode : outBytecode) {
-            printf("%.2X ", bytecode);
+                  std::ostreambuf_iterator<char>(outFile));
+        
+        if (verbose == true) {
```
Comment (rwblickhan) [inline src/main.cpp]: This could be simplified to `if (verbose) {`.

**Review by rwblickhan — APPROVED** [review]

File: src/main.cpp (hunk 2 above)
Comment (bobheadxi) [reply src/main.cpp:41]: In-house CPP argparse library? 😮

File: src/main.cpp (hunk 3 above)
Comment (bobheadxi) [reply src/main.cpp]: Do you have any functionality in mind right now for `-v` for the others? I can put them in this PR as well

File: src/main.cpp (hunk 2 above)
Comment (rwblickhan) [reply src/main.cpp:41]: Haha that's almost what we have already 😛 But for this I specifically meant the fact that we're hardcoding all the `args.back() != ...` lines.

**Comment by rwblickhan** [comment]
This can be rebased now that the `compiler` branch has been merged.

**Comment by rwblickhan** [comment]
Okay, I cherry-picked instead. I think I got all the relevant parts.

**Pull request merged**